discovery_engine_core/web-api/openapi.yaml

```diff
@@ -454,16 +461,20 @@ components:
       allOf:
         - $ref: '#/components/schemas/BaseError'
         - type: object
-          required: [documents]
+          required: [details]
           properties:
-            documents:
-              type: array
-              items:
-                type: object
-                required: [id]
-                properties:
-                  id:
-                    $ref: '#/components/schemas/DocumentId'
+            details:
+              type: object
```

Comment:
:grey_question: why is it necessary to introduce an extra level of indirection in `details`?

Reply:
It makes a lot of things simpler:

- keep the fields of the base error and specific errors separate, so accidental field overlaps are impossible
- allows easily to specify (in openapi.yaml) a field for more details as an free-form object (as we don't want to specify additional error details which are not meant to be processed by a client but are purely informational for a user debugging issues; the reason we don't want to specify them is so that they can freely change in both availability and formatting)
- have a easy "cut" point where we can reduce the amount of details an error contains for fully public production deployments
- simpler to implement in code